### ISSUE TITLE
fix(testing): add rmsd wrappers to FrameworkAdapter and test wrt=Q gradient

### ISSUE THREAD

**Issue by hunter-heidenreich** [hunter-heidenreich/Kabsch-Cookbook#18]
### 1. Inline adapter dispatch repeated 3 times

```python
adapter_cls = type(adapter).__name__
if adapter_cls == "PyTorchAdapter":
    rmsd_wrapper = kabsch_torch.kabsch_rmsd(P, Q)
elif adapter_cls == "JAXAdapter":
    rmsd_wrapper = kabsch_jax.kabsch_rmsd(P, Q)
elif adapter_cls == "TFAdapter":
    ...
elif adapter_cls == "MLXAdapter":
    ...
else:
    pytest.skip(...)
```

This pattern is copy-pasted across 3 test methods. Adding a new framework requires edits in 3 places and risks being missed.

`FrameworkAdapter` already exposes all other public API methods (`kabsch`, `horn`, `kabsch_umeyama`, etc.) as adapter methods. `kabsch_rmsd` and `kabsch_umeyama_rmsd` should be added to `FrameworkAdapter` and all subclasses (`PyTorchAdapter`, `JAXAdapter`, `TFAdapter`, `MLXAdapter`), then the test methods should call `adapter.kabsch_rmsd(P, Q)` directly.

### 2. RMSD wrapper gradient only tested for `wrt="P"`

```python
grad = adapter.get_grad(P, Q, wrapper, seed=None, wrt="P")
```

`wrt="Q"` is never tested for the RMSD wrappers. The `wrt="Q"` backward path through `kabsch_rmsd` / `kabsch_umeyama_rmsd` is untested for gradient finiteness.
